riscv/csrs.cc

```diff
@@ -455,6 +455,10 @@ sstatus_proxy_csr_t::sstatus_proxy_csr_t(processor_t* const proc, const reg_t ad
 bool sstatus_proxy_csr_t::unlogged_write(const reg_t val) noexcept {
   const reg_t new_mstatus = (mstatus->read() & ~sstatus_write_mask) | (val & sstatus_write_mask);
 
+  // On RV32 this will only log the low 32 bits, so make sure we're
+  // not modifying anything in the upper 32 bits.
+  assert((sstatus_write_mask & 0xffffffffU) == sstatus_write_mask);
+
   mstatus->write(new_mstatus);
```

Comment:
Is this necessary? sstatus_write_mask seems to be a nearly const value calculated by compute_sstatus_write_mask:
```
  return 0
    | (proc->extension_enabled('S') ? (SSTATUS_SIE | SSTATUS_SPIE | SSTATUS_SPP) : 0)
    | (has_page ? (SSTATUS_SUM | SSTATUS_MXR) : 0)
    | (has_fs ? SSTATUS_FS : 0)
    | (proc->any_custom_extensions() ? SSTATUS_XS : 0)
    | (has_vs ? SSTATUS_VS : 0)
    ;
```

Reply:
The `sstatus` write mask is computed a mile away and can change in the future. The assumption in this code (about none of the upper bits changing) is not at all obvious, so I wanted to document and enforce it with the assertion.

One alternative is to move the assertion into `compute_sstatus_write_mask()` so it's only executed once, and update the comment to explain specifically what code is reliant on this assumption.

What do you think?

Reply:
OK. I see. Just leave it here.